tests/integration_tests/test_grpc_only.py

```diff
@@ -58,33 +60,43 @@ def test_grpc_mode(custom_ethermint):
         "to": contract.address,
         "data": contract.encodeABI(fn_name="currentChainID"),
     }
-    api_port = ports.api_port(custom_ethermint.base_port(2))
+    api_port = ports.api_port(custom_ethermint.base_port(1))
     # in normal mode, grpc query works even if we don't pass chain_id explicitly
-    rsp = grpc_eth_call(api_port, msg)
-    print(rsp)
-    assert "code" not in rsp, str(rsp)
-    assert 9000 == int.from_bytes(base64.b64decode(rsp["ret"].encode()), "big")
-
+    success = False
+    for i in range(3):
+        rsp = grpc_eth_call(api_port, msg)
+        print(i, rsp)
+        assert "code" not in rsp, str(rsp)
+        ret = rsp["ret"]
+        valid = ret is not None
+        if valid and 9000 == int.from_bytes(base64.b64decode(ret.encode()), "big"):
+            success = True
+            break
+        time.sleep(1)
+    assert success
+    # wait 1 block before stop
+    for i in range(2):
```

Comment:
I don't understand this iteration

Reply:
wait 1 more block to avoid node stopped before tnx included

Reply:
I understand that part, but I don't understand why we need the iteration

Reply:
since I cant predict which node proposed the block that include this tnx